src/main/java/org/sagebionetworks/web/client/widget/footer/FooterViewImpl.java

```diff
@@ -1,163 +1,50 @@
-
-  @UiField
-  Anchor copyrightYear2;
-
-  @UiField
-  Span portalVersionSpan;
-
-  @UiField
-  Span srcVersionSpan;
-
-  @UiField
-  Span srcVersionSpan2;
-
-  @UiField
-  Span portalVersionSpan2;
-
-  @UiField
-  Span repoVersionSpan;
-
-  @UiField
-  Span repoVersionSpan2;
-
-  @UiField
-  Anchor reportAbuseLink;
-
-  @UiField
-  Anchor reportAbuseLink2;
-
-  @UiField
-  Span hideACTActionsContainer;
-
-  @UiField
-  Image alphaModeStatusIcon;
-
-  @UiField
-  FocusPanel sageLogo1;
-
-  @UiField
-  FocusPanel sageLogo2;
-
-  SageImageBundle sageImageBundle;
   String portalVersion, repoVersion, srcVersion;
-  private Presenter presenter;
-  private CookieProvider cookies;
-  private GlobalApplicationState globalAppState;
-  private ToggleACTActionsButton hideACTActionsButton;
-  Div container = new Div();
+  PortalGinInjector ginInjector;
+  SynapseReactClientFullContextPropsProvider propsProvider;
 
   @Inject
   public FooterViewImpl(
-    Binder binder,
-    CookieProvider cookies,
-    GlobalApplicationState globalAppState,
-    ToggleACTActionsButton hideACTActionsButton,
+    PortalGinInjector ginInjector,
     GWTWrapper gwt,
-    SageImageBundle sageImageBundle
+    SynapseReactClientFullContextPropsProvider propsProvider
   ) {
-    this.sageImageBundle = sageImageBundle;
-
+    this.ginInjector = ginInjector;
+    this.propsProvider = propsProvider;
     // defer constructing this view (to give a chance for other page components to load first)
     Callback constructViewCallback = () -> {
-      IsWidget widget = binder.createAndBindUi(this);
-      container.add(widget);
-      initDebugModeLink();
-      hideACTActionsContainer.add(hideACTActionsButton);
-      copyrightYear.setText(
-        DateTimeFormat.getFormat("yyyy").format(new Date())
-      );
-      copyrightYear2.setText(
-        DateTimeFormat.getFormat("yyyy").format(new Date())
-      );
-      reportAbuseLink.addClickHandler(event -> {
-        presenter.onReportAbuseClicked();
-      });
-      reportAbuseLink2.addClickHandler(event -> {
-        presenter.onReportAbuseClicked();
-      });
-      if (portalVersion != null) {
-        portalVersionSpan.setText(portalVersion);
-        portalVersionSpan2.setText(portalVersion);
-        repoVersionSpan.setText(repoVersion);
-        repoVersionSpan2.setText(repoVersion);
-        srcVersionSpan.setText(srcVersion);
-        srcVersionSpan2.setText(srcVersion);
-      }
-      ClickHandler sageLogoClickHandler = event -> {
-        Window.Location.assign("https://sagebionetworks.org");
-      };
-      sageLogo1.addClickHandler(sageLogoClickHandler);
-      sageLogo2.addClickHandler(sageLogoClickHandler);
+      wrapper.add(container);
       refresh();
     };
-    gwt.scheduleExecution(constructViewCallback, 2500);
-    this.cookies = cookies;
-    this.globalAppState = globalAppState;
-    this.hideACTActionsButton = hideACTActionsButton;
+    gwt.scheduleExecution(constructViewCallback, 2000);
```

Comment:
just curious, why do we delay initializing the footer?

Reply:
I think it was introduced to delay working on something that should be off screen, and because some page content doesn't push the footer down until loaded (so the footer would initially appear towards the top of the page, and eventually be pushed to the bottom)